From 8c22e064388c3d46c461ffc7847cae2dfab18ce2 Mon Sep 17 00:00:00 2001
From: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Date: Tue, 30 Jun 2020 11:24:24 +0200
Subject: [PATCH 2/5] build/meson: disable qat_asym driver

QAT docs in DPDK say:
[...] for crypto, where each QAT VF may expose two crypto devices,
sym and asym, it may happen that the number of devices will be bigger
than MAX_DEVS and the process will show an error during PMD
initialisation. To avoid this problem CONFIG_RTE_CRYPTO_MAX_DEVS may
be increased or -w, pci-whitelist domain:bus:devid:func option may be
used.

We run into this limitation in our CI. We didn't compile the asym driver
with Makefiles, so don't compile it with meson either. Increasing
MAX_DEVS unnecessarily increases the total memory footprint, so we
don't go that way.

Change-Id: Idf27335ef0c8890727909985bd3e909edb1282e6
Signed-off-by: Darek Stojaczyk <dariusz.stojaczyk@intel.com>
Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3517 (spdk-20.05)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/3770 (spdk-20.08)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/5338 (spdk-20.11)
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/6428 (spdk-21.02)
Signed-off-by: Krzysztof Karas <krzysztof.karas@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/8136
Reviewed-by: Aleksey Marchuk <alexeymar@mellanox.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
---
 drivers/common/qat/meson.build | 4 ++--
 drivers/crypto/qat/meson.build | 3 ---
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/common/qat/meson.build b/drivers/common/qat/meson.build
index 479a46f..d695f7d 100644
--- a/drivers/common/qat/meson.build
+++ b/drivers/common/qat/meson.build
@@ -64,10 +64,10 @@ endif
 
 if qat_crypto
     foreach f: ['qat_sym_pmd.c', 'qat_sym.c', 'qat_sym_session.c',
-            'qat_sym_hw_dp.c', 'qat_asym_pmd.c', 'qat_asym.c']
+            'qat_sym_hw_dp.c']
         sources += files(join_paths(qat_crypto_relpath, f))
     endforeach
     deps += ['security']
     ext_deps += libcrypto
-    cflags += ['-DBUILD_QAT_SYM', '-DBUILD_QAT_ASYM']
+    cflags += ['-DBUILD_QAT_SYM']
 endif
diff --git a/drivers/crypto/qat/meson.build b/drivers/crypto/qat/meson.build
index b3b2d17..195a4f5 100644
--- a/drivers/crypto/qat/meson.build
+++ b/drivers/crypto/qat/meson.build
@@ -13,8 +13,6 @@ qat_deps += 'security'
 if dep.found()
     # Add our sources files to the list
     qat_sources += files(
-            'qat_asym.c',
-            'qat_asym_pmd.c',
             'qat_sym.c',
             'qat_sym_hw_dp.c',
             'qat_sym_pmd.c',
@@ -22,5 +20,4 @@ if dep.found()
 	)
     qat_ext_deps += dep
     qat_cflags += '-DBUILD_QAT_SYM'
-    qat_cflags += '-DBUILD_QAT_ASYM'
 endif
-- 
1.8.3.1

